src/polynomial.cpp

```diff
@@ -0,0 +1,307 @@
+#include "polynomial.h"
+#include "add.h"
+#include "mul.h"
+#include "pow.h"
+#include "constants.h"
+
+namespace SymEngine {
+
+Polynomial::Polynomial(const RCP<const Symbol> &var, map_uint_mpz&& dict) :
+     var_{var}, dict_{std::move(dict)} {
+
+    map_uint_mpz::iterator it = dict_.begin();
+    uint largest = it->first;
+    uint cur;
+    while (it != dict_.end()) {
+        cur =it->first;
+        if (cur > largest)
+            largest = cur;
+        it++;
+    }
+    degree = largest;
```

Comment:
It can be passed as a parameter, but in many internal calls then we have to calculate degree from dict, then pass. Won't this be an overhead?